components/motor/dimensionengineering/sabertooth.go

```diff
@@ -0,0 +1,556 @@
+type controller struct {
+	mu           sync.Mutex
+	port         io.ReadWriteCloser
+	serialDevice string
+	logger       golog.Logger
+	activeAxes   map[int]bool
+	testChan     chan []byte
+	address      int // 128-135
+}
+
+// Motor is a single axis/motor/component instance.
+type Motor struct {
+	// A reference to the actual controller that needs to be commanded for the motor to run
+	c *controller
+	// which channel the motor is connected to on the controller
+	Channel int
+	// Simply indicates if the RDK _thinks_ the motor is moving, because this controller has no feedback, this may not reflect reality
+	isOn bool
+	// dirFlip means that the motor is wired "backwards" from what we expect forward/backward to mean,
+	// so we need to "flip" the direction sent by control
+	dirFlip bool
+	// the minimum power that can be set for the motor to prevent stalls
+	minPowerPct float64
+	// the maximum power that can be set for the motor
+	maxPowerPct float64
+	// the freewheel RPM of the motor
+	maxRPM float64
+
+	// A manager to ensure only a single operation is happening at any given time since commands could overlap on the serial port
+	opMgr operation.SingleOperationManager
+}
+
+// Config adds DimensionEngineering-specific config options.
+type Config struct {
+	// path to /dev/ttyXXXX file
+	SerialPath string `json:"serial_path"`
+
+	// The baud rate of the controller
+	BaudRate int `json:"serial_baud_rate,omitempty"`
+
+	// Valid values are 128-135
+	SerialAddress int `json:"serial_address"`
+
+	// Valid values are 1/2
+	MotorChannel int `json:"motor_channel"`
+
+	// Flip the direction of the signal sent to the controller.
+	// Due to wiring/motor orientation, "forward" on the controller may not represent "forward" on the robot
+	DirectionFlip bool `json:"dir_flip,omitempty"`
+
+	// A value to control how quickly the controller ramps to a particular setpoint
+	RampValue int `json:"controller_ramp_value,omitempty"`
+
+	// The maximum freewheel rotational velocity of the motor after the final drive (maximum effective wheel speed)
+	MaxRPM float64 `json:"max_rpm,omitempty"`
+
+	// The name of the encoder used for this motor
+	Encoder string `json:"encoder,omitempty"`
+
+	// The lowest power percentage to allow for this motor. This is used to prevent motor stalls and overheating. Default is 0.0
+	MinPowerPct float64 `json:"min_power_pct,omitempty"`
+
+	// The max power percentage to allow for this motor. Default is 0.0
+	MaxPowerPct float64 `json:"max_power_pct,omitempty"`
+
+	// The number of ticks per rotation of this motor from the encoder
+	TicksPerRotation int `json:"ticks_per_rotation,omitempty"`
+
+	// TestChan is a fake "serial" path for test use only
+	TestChan chan []byte `json:"-,omitempty"`
+}
+
+// Validate ensures all parts of the config are valid.
+func (cfg *Config) Validate(path string) error {
+	if cfg.SerialPath == "" {
+		return utils.NewConfigValidationFieldRequiredError(path, "serial_path")
```

Comment:
you deny the config when path is not mentioned but in your constructor you allow for an empty path? Should we reject the config in this case?